### DISCUSSION TITLE
ReplicatorConfiguration filter example

### DISCUSSION THREAD

**Discussion by filipenanclarez** [cofu-app/cbl-dart#314] [Q&A] [answered]
Hi, 

How can'i filter documents in Replicator?

Can you show me a example with ReplicatorConfiguration to replicate only a group of documents?

**Comment by blaugold** [comment, marked as answer]
Hey @filipenanclarez,

you've got two options for filtering.

1. Specify a list of IDs of the documents that should be replicated:

```dart
ReplicatorConfiguration(
  ...
  documentIds: [
    "foo",
    "bar",
  ],
  ...
);
```

With this configuration, only the documents with the IDs `foo` and `bar` are replicated. This works both for push and pull replication.

This also works for documents that are not in the local database yet. For example, if you got a document ID from another document, but the first document is not yet in the local database, you can run the replicator in pull mode and only with the ID of that document to fetch it.

2. Provide a function that is called for each document and decides whether a document should be replicated:

```dart
ReplicatorConfiguration(
  ...
  pushFilter: (doc, flags) {
    return !doc.boolean("isLocal");
  }
  ...
);
```

With this configuration, documents are only pushed if they don't have the property `isLocal` or where `isLocal` is `false`.

There is an equivalent option, `pullFilter`, with the same signature for filtering which documents should be pulled.

Does this help?
